src/lib/components/Form/service/ServiceForm.svelte

```diff
@@ -48,6 +48,13 @@
           featureTypeCheckmarkVisible = true;
         } else if (key === 'downloads') {
           downloadCheckmarkVisible = true;
+        } else if (key === 'legendImage') {
+          // legend sizes are determined and returned in the backend
+          const json = await response.json();
+          value = json.isoMetadata.services.find(
+            (s: Service) => s.serviceIdentification === service.serviceIdentification
+          ).legendImage;
```

Comment:
```suggestion
          value = json.isoMetadata.services?.find(
            (s: Service) => s.serviceIdentification === service.serviceIdentification
          )?.legendImage;
```

This might fail when creating a new metadata set so i suggest to add some `?`